sidecar_ecs/sidecar_ecs_resources.tf

```diff
@@ -1,3 +1,7 @@
+locals {
+  selected_subnets = coalesce(var.subnet_ids, data.aws_subnets.all_subnets.ids)
```

Comment:
I think we should remove the data source completely. We should not assume subnets and just use those provided by the user.

Reply:
agreed!